contracts/pool/pools/liquid/WithdrawalEligibility.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.18;
+
+import "../TokenPoolBase.sol";
+
+error UnlockPeriodNotFinished();
+error NonWithdrawableTokens();
+error AmountExceedsAvailableWithdraw();
+error StakedAmountExceeded();
+error LockedAmountExceeded();
+
+/**
+ * @title WithdrawalEligibility
+ * @dev Provides functionality for checking withdrawal eligibility and grace periods
+ */
+contract WithdrawalEligibility is TokenPoolBase {
+    uint256 internal constant GRACE_PERIOD = 4 weeks;
+    uint256 internal constant GRACE_PERIOD_REMOVAL_FEE = 1000;
+    uint256 internal constant UPGRADE_PERIOD = 63_158_400; // 2 years considering leap years in seconds;
+
+    struct EligibleWithdraw {
+        uint256 gracePeriodEndTime;
+        uint256 amount;
+    }
+
+    struct Stake {
+        EligibleWithdraw eligibleWithdraw;
+        uint256 amount;
+    }
+
+    struct Lock {
+        EligibleWithdraw eligibleWithdraw;
+        uint256 lockTime;
```

Comment:
Mislim da se ova 2 podatka trenutno ne mogu dohvatit izvan contracta za nekog usera u sklopu mappinga `WithdrawalEligibility::locks`, pa samo provjeri je li nam to okej.